docs/simulation_development.md

```diff
@@ -0,0 +1,28 @@
+# Adding new performance testing simulations
+
+## Simulation coding guideline
+- New simulations can be added to the `./performance-test/src/gatling/java/org/opensearch/dataprepper/test/performance` folder.
+- Simulations are written in Java.
+- Use a class name that describes what type of scenario your simulation is designed to simulate and end with a `Simulation` suffix. 
+- Simulations are executed using the Simulations classes static function.
+- Test results must be repeatable. No element of the test should depend on a random element.
```

Comment:
Could you specify the scope of random elements? AFAICT, there could be random elements in test data, e.g.
```
final String log = "{\"log\": \"127.0.0.1 - frank [" + now() + "] \\\"GET /apache_pb.gif HTTP/1.0\\\" 200 2326\"}";
```